### ISSUE TITLE
errors in v0.8.1

### ISSUE THREAD

**Issue by sunyi000** [pyapp-kit/psygnal#195]
* psygnal version: 0.8..1
* Python version: 3.8
* Operating System: Ubuntu

### Description

Recently we @tischi rebuilt our conda environment, and as a dependency of one of the packages(napari), psygnal 0.8.1 is automatically installed in the environment. It caused the below error.

```
File "/opt/conda/envs/napari/lib/python3.8/site-packages/magicgui/widgets/bases/__init__.py", line 45, in <module>
    from ._button_widget import ButtonWidget
  File "/opt/conda/envs/napari/lib/python3.8/site-packages/magicgui/widgets/bases/_button_widget.py", line 5, in <module>
    from psygnal import Signal, SignalInstance
  File "/opt/conda/envs/napari/lib/python3.8/site-packages/psygnal/__init__.py", line 58, in <module>
    from ._evented_decorator import evented, get_evented_namespace, is_evented
ImportError: cannot import name 'get_evented_namespace' from 'psygnal._evented_decorator' (/opt/conda/envs/napari/lib/python3.8/site-packages/psygnal/_evented_decorator.cpython-38-x86_64-linux-gnu.so)
```

This error disappears if we revert psygnal to version 0.7.0 or 0.7.2 . Is this an expected behaviour for the new  psygnal version?

Thanks

**Comment by tlambert03** [comment]
thanks!  can you tell whether you installed from conda or pip?

**Comment by sunyi000** [comment]
> 
Thank you
we installed via pip install napari[all], but napari-ome-zarr plugin is installed via conda-forge

**Comment by tlambert03** [comment]
thanks again, if you have a moment, please give me the output of `conda list` (I'm struggling to reproduce it, so still trying to figure out what's going on here... haven't tried ubuntu yet though)

**Comment by sunyi000** [comment]
here is what we did.
1. create a napari conda env with python 3.8
2. pip install napari[all]
3. conda install -n napari  -c conda-forge napari-ome-zarr cudatoolkit=11.2
4. pip install stardist-napari
5. pip install PyQt5==5.12.3 (the newest pyqt caused an error, so we switch it back to 5.12.3)
6. pip install psygnal==0.7.2

The output of conda list

```
WARNING conda.gateways.disk.delete:unlink_or_rename_to_trash(143): Could not remove or rename /opt/conda/envs/napari/conda-meta/pyqt-5.15.7-py38ha0d8c90_3.json.  Please remove this file manually (you may need to reboot to free file handles)
WARNING conda.gateways.disk.delete:unlink_or_rename_to_trash(143): Could not remove or rename /opt/conda/envs/napari/conda-meta/sphinx-6.1.3-pyhd8ed1ab_0.json.  Please remove this file manually (you may need to reboot to free file handles)
WARNING conda.gateways.disk.delete:unlink_or_rename_to_trash(143): Could not remove or rename /opt/conda/envs/napari/conda-meta/scipy-1.10.1-py38h10c12cc_0.json.  Please remove this file manually (you may need to reboot to free file handles)
WARNING conda.gateways.disk.delete:unlink_or_rename_to_trash(143): Could not remove or rename /opt/conda/envs/napari/conda-meta/docutils-0.19-py38h578d9bd_1.json.  Please remove this file manually (you may need to reboot to free file handles)
# packages in environment at /opt/conda/envs/napari:
#
# Name                	Version               	Build  Channel
_libgcc_mutex         	0.1             	conda_forge	conda-forge
_openmp_mutex         	4.5                   	2_gnu	conda-forge
absl-py               	1.4.0                	pypi_0	pypi
aiohttp               	3.8.4        	py38h1de0b5d_0	conda-forge
aiosignal             	1.3.1          	pyhd8ed1ab_0	conda-forge
alabaster             	0.7.13         	pyhd8ed1ab_0	conda-forge
alsa-lib              	1.2.8            	h166bdaf_0	conda-forge
aom                   	3.5.0            	h27087fc_0	conda-forge
app-model             	0.1.1                	pypi_0	pypi
appdirs               	1.4.4          	pyh9f0ad1d_0	conda-forge
asciitree             	0.3.3                  	py_2	conda-forge
asttokens             	2.2.1          	pyhd8ed1ab_0	conda-forge
astunparse            	1.6.3                	pypi_0	pypi
async-timeout         	4.0.2          	pyhd8ed1ab_0	conda-forge
attr                  	2.5.1            	h166bdaf_1	conda-forge
attrs                 	22.2.0         	pyh71513ae_0	conda-forge
babel                 	2.12.1         	pyhd8ed1ab_0	conda-forge
backcall              	0.2.0          	pyh9f0ad1d_0	conda-forge
backports             	1.0            	pyhd8ed1ab_3	conda-forge
backports.functools_lru_cache 1.6.4          	pyhd8ed1ab_0	conda-forge
blosc                 	1.21.3           	hafa529b_0	conda-forge
bokeh                 	2.4.3          	pyhd8ed1ab_3	conda-forge
brotli                	1.0.9            	h166bdaf_8	conda-forge
brotli-bin            	1.0.9            	h166bdaf_8	conda-forge
brotlipy              	0.7.0       	py38h0a891b7_1005	conda-forge
brunsli               	0.1              	h9c3ff4c_0	conda-forge
bzip2                 	1.0.8            	h7f98852_4	conda-forge
c-ares                	1.18.1           	h7f98852_0	conda-forge
c-blosc2              	2.7.1            	hf91038e_0	conda-forge
ca-certificates       	2022.12.7        	ha878542_0	conda-forge
cachetools            	5.3.0                	pypi_0	pypi
cachey                	0.2.1          	pyh9f0ad1d_0	conda-forge
cairo                 	1.16.0        	ha61ee94_1014	conda-forge
certifi               	2022.12.7            	pypi_0	pypi
cffi                  	1.15.1       	py38h4a40e3a_3	conda-forge
cfitsio               	4.2.0            	hd9d235c_0	conda-forge
charls                	2.4.1            	hcb278e6_0	conda-forge
charset-normalizer    	3.0.1                	pypi_0	pypi
click                 	8.1.3       	unix_pyhd8ed1ab_2	conda-forge
cloudpickle           	2.2.1          	pyhd8ed1ab_0	conda-forge
colorama              	0.4.6          	pyhd8ed1ab_0	conda-forge
comm                  	0.1.2          	pyhd8ed1ab_0	conda-forge
contourpy             	1.0.7                	pypi_0	pypi
cryptography          	39.0.1       	py38h3d167d9_0	conda-forge
csbdeep               	0.7.3                	pypi_0	pypi
cudatoolkit           	11.2.2          	hbe64b41_11	conda-forge
cycler                	0.11.0               	pypi_0	pypi
cytoolz               	0.12.0       	py38h0a891b7_1	conda-forge
dask                  	2023.2.1       	pyhd8ed1ab_0	conda-forge
dask-core             	2023.2.1       	pyhd8ed1ab_0	conda-forge
dav1d                 	1.0.0            	h166bdaf_1	conda-forge
dbus                  	1.13.6           	h5008d03_3	conda-forge
debugpy               	1.6.6        	py38h8dc9893_0	conda-forge
decorator             	5.1.1          	pyhd8ed1ab_0	conda-forge
distributed           	2023.2.1       	pyhd8ed1ab_0	conda-forge
docstring_parser      	0.15           	pyhd8ed1ab_0	conda-forge
docutils              	0.17.1               	pypi_0	pypi
entrypoints           	0.4            	pyhd8ed1ab_0	conda-forge
executing             	1.2.0          	pyhd8ed1ab_0	conda-forge
expat                 	2.5.0            	h27087fc_0	conda-forge
fasteners             	0.17.3         	pyhd8ed1ab_0	conda-forge
fftw                  	3.3.10      	nompi_hf0379b8_106	conda-forge
flatbuffers           	23.1.21              	pypi_0	pypi
font-ttf-dejavu-sans-mono 2.37             	hab24e00_0	conda-forge
font-ttf-inconsolata  	3.000            	h77eed37_0	conda-forge
font-ttf-source-code-pro  2.038            	h77eed37_0	conda-forge
font-ttf-ubuntu       	0.83             	hab24e00_0	conda-forge
fontconfig            	2.14.2           	h14ed4e7_0	conda-forge
fonts-conda-ecosystem 	1                         	0	conda-forge
fonts-conda-forge     	1                         	0	conda-forge
fonttools             	4.38.0               	pypi_0	pypi
freetype              	2.12.1           	hca18f0e_1	conda-forge
freetype-py           	2.3.0          	pyhd8ed1ab_0	conda-forge
frozenlist            	1.3.3        	py38h0a891b7_0	conda-forge
fsspec                	2023.1.0       	pyhd8ed1ab_0	conda-forge
gast                  	0.4.0                	pypi_0	pypi
gettext               	0.21.1           	h27087fc_0	conda-forge
giflib                	5.2.1            	h36c2ea0_2	conda-forge
glib                  	2.74.1           	h6239696_1	conda-forge
glib-tools            	2.74.1           	h6239696_1	conda-forge
google-auth           	2.16.1               	pypi_0	pypi
google-auth-oauthlib  	0.4.6                	pypi_0	pypi
google-pasta          	0.2.0                	pypi_0	pypi
graphite2             	1.3.13        	h58526e2_1001	conda-forge
grpcio                	1.51.3               	pypi_0	pypi
gst-plugins-base      	1.22.0           	h4243ec0_0	conda-forge
gstreamer             	1.22.0           	h25f0c4b_0	conda-forge
gstreamer-orc         	0.4.33           	h166bdaf_0	conda-forge
h5py                  	3.8.0                	pypi_0	pypi
harfbuzz              	6.0.0            	h8e241bc_0	conda-forge
heapdict              	1.0.1                  	py_0	conda-forge
hsluv                 	5.0.3                	pypi_0	pypi
icu                   	70.1             	h27087fc_0	conda-forge
idna                  	3.4            	pyhd8ed1ab_0	conda-forge
imagecodecs           	2023.1.23    	py38h3ca0a39_0	conda-forge
imageio               	2.26.0         	pyh24c5eb1_0	conda-forge
imagesize             	1.4.1          	pyhd8ed1ab_0	conda-forge
importlib-metadata    	6.0.0          	pyha770c72_0	conda-forge
importlib_metadata    	6.0.0            	hd8ed1ab_0	conda-forge
importlib_resources   	5.12.0         	pyhd8ed1ab_0	conda-forge
in-n-out              	0.1.7                	pypi_0	pypi
ipykernel             	6.21.2         	pyh210e3f2_0	conda-forge
ipython               	8.11.0         	pyh41d4057_0	conda-forge
ipython-genutils      	0.2.0                	pypi_0	pypi
ipython_genutils      	0.2.0                  	py_1	conda-forge
jack                  	1.9.22           	h11f4161_0	conda-forge
jedi                  	0.18.2         	pyhd8ed1ab_0	conda-forge
jinja2                	3.1.2          	pyhd8ed1ab_1	conda-forge
jpeg                  	9e               	h0b41bf4_3	conda-forge
jsonschema            	4.17.3         	pyhd8ed1ab_0	conda-forge
jupyter_client        	8.0.3          	pyhd8ed1ab_0	conda-forge
jupyter_core          	5.2.0        	py38h578d9bd_0	conda-forge
jxrlib                	1.1              	h7f98852_2	conda-forge
keras                 	2.11.0               	pypi_0	pypi
keyutils              	1.6.1            	h166bdaf_0	conda-forge
kiwisolver            	1.4.4        	py38h43d8883_1	conda-forge
krb5                  	1.20.1           	h81ceb04_0	conda-forge
lame                  	3.100         	h166bdaf_1003	conda-forge
lazy-loader           	0.1                  	pypi_0	pypi
lcms2                 	2.14             	hfd0df8a_1	conda-forge
ld_impl_linux-64      	2.40             	h41732ed_0	conda-forge
lerc                  	4.0.0            	h27087fc_0	conda-forge
libaec                	1.0.6            	hcb278e6_1	conda-forge
libavif               	0.11.1           	h5cdd6b5_0	conda-forge
libblas               	3.9.0       	16_linux64_openblas	conda-forge
libbrotlicommon       	1.0.9            	h166bdaf_8	conda-forge
libbrotlidec          	1.0.9            	h166bdaf_8	conda-forge
libbrotlienc          	1.0.9            	h166bdaf_8	conda-forge
libcap                	2.66             	ha37c62d_0	conda-forge
libcblas              	3.9.0       	16_linux64_openblas	conda-forge
libclang              	15.0.6.1             	pypi_0	pypi
libclang13            	15.0.7      	default_h3e3d535_1	conda-forge
libcups               	2.3.3            	h36d4200_3	conda-forge
libcurl               	7.88.1           	hdc1c0ab_0	conda-forge
libdb                 	6.2.32           	h9c3ff4c_0	conda-forge
libdeflate            	1.17             	h0b41bf4_0	conda-forge
libedit               	3.1.20191231     	he28a2e2_2	conda-forge
libev                 	4.33             	h516909a_1	conda-forge
libevent              	2.1.10           	h28343ad_4	conda-forge
libffi                	3.4.2            	h7f98852_5	conda-forge
libflac               	1.4.2            	h27087fc_0	conda-forge
libgcc-ng             	12.2.0          	h65d4601_19	conda-forge
libgcrypt             	1.10.1           	h166bdaf_0	conda-forge
libgfortran-ng        	12.2.0          	h69a702a_19	conda-forge
libgfortran5          	12.2.0          	h337968e_19	conda-forge
libglib               	2.74.1           	h606061b_1	conda-forge
libgomp               	12.2.0          	h65d4601_19	conda-forge
libgpg-error          	1.46             	h620e276_0	conda-forge
libiconv              	1.17             	h166bdaf_0	conda-forge
liblapack             	3.9.0       	16_linux64_openblas	conda-forge
libllvm15             	15.0.7           	hadd5161_0	conda-forge
libnghttp2            	1.52.0           	h61bc06f_0	conda-forge
libnsl                	2.0.0            	h7f98852_0	conda-forge
libogg                	1.3.4            	h7f98852_1	conda-forge
libopenblas           	0.3.21      	pthreads_h78a6416_3	conda-forge
libopus               	1.3.1            	h7f98852_1	conda-forge
libpng                	1.6.39           	h753d276_0	conda-forge
libpq                 	15.2             	hb675445_0	conda-forge
libsndfile            	1.2.0            	hb75c966_0	conda-forge
libsodium             	1.0.18           	h36c2ea0_1	conda-forge
libsqlite             	3.40.0           	h753d276_0	conda-forge
libssh2               	1.10.0           	hf14f497_3	conda-forge
libstdcxx-ng          	12.2.0          	h46fd767_19	conda-forge
libsystemd0           	252              	h2a991cd_0	conda-forge
libtiff               	4.5.0            	h6adf6a1_2	conda-forge
libtool               	2.4.7            	h27087fc_0	conda-forge
libudev1              	253              	h0b41bf4_0	conda-forge
libuuid               	2.32.1        	h7f98852_1000	conda-forge
libvorbis             	1.3.7            	h9c3ff4c_0	conda-forge
libwebp-base          	1.2.4            	h166bdaf_0	conda-forge
libxcb                	1.13          	h7f98852_1004	conda-forge
libxkbcommon          	1.5.0            	h79f4944_0	conda-forge
libxml2               	2.10.3           	h7463322_0	conda-forge
libzlib               	1.2.13           	h166bdaf_4	conda-forge
libzopfli             	1.0.3            	h9c3ff4c_0	conda-forge
llvmlite              	0.39.1               	pypi_0	pypi
locket                	1.0.0          	pyhd8ed1ab_0	conda-forge
lz4                   	4.3.2        	py38hd012fdc_0	conda-forge
lz4-c                 	1.9.4            	hcb278e6_0	conda-forge
magicgui              	0.7.0          	pyhd8ed1ab_0	conda-forge
markdown              	3.4.1                	pypi_0	pypi
markdown-it-py        	2.2.0          	pyhd8ed1ab_0	conda-forge
markupsafe            	2.1.2        	py38h1de0b5d_0	conda-forge
matplotlib            	3.7.0                	pypi_0	pypi
matplotlib-inline     	0.1.6          	pyhd8ed1ab_0	conda-forge
mdurl                 	0.1.2                	pypi_0	pypi
mpg123                	1.31.2           	hcb278e6_0	conda-forge
msgpack-python        	1.0.4        	py38h43d8883_1	conda-forge
multidict             	6.0.4        	py38h1de0b5d_0	conda-forge
mypy_extensions       	1.0.0          	pyha770c72_0	conda-forge
mysql-common          	8.0.32           	ha901b37_0	conda-forge
mysql-libs            	8.0.32           	hd7da12d_0	conda-forge
napari                	0.4.17               	pypi_0	pypi
napari-console        	0.0.7          	pyhd8ed1ab_0	conda-forge
napari-ome-zarr       	0.5.2          	pyhd8ed1ab_0	conda-forge
napari-plugin-engine  	0.2.0          	pyhd8ed1ab_2	conda-forge
napari-svg            	0.1.6          	pyhd8ed1ab_1	conda-forge
ncurses               	6.3              	h27087fc_1	conda-forge
nest-asyncio          	1.5.6          	pyhd8ed1ab_0	conda-forge
networkx              	3.0            	pyhd8ed1ab_0	conda-forge
npe2                  	0.6.2          	pyhd8ed1ab_0	conda-forge
nspr                  	4.35             	h27087fc_0	conda-forge
nss                   	3.88             	he45b914_0	conda-forge
numba                 	0.56.4               	pypi_0	pypi
numcodecs             	0.11.0       	py38h8dc9893_1	conda-forge
numpy                 	1.23.5               	pypi_0	pypi
numpydoc              	1.5.0          	pyhd8ed1ab_0	conda-forge
oauthlib              	3.2.2                	pypi_0	pypi
ome-zarr              	0.6.1          	pyhd8ed1ab_0	conda-forge
openjpeg              	2.5.0            	hfec8fc6_2	conda-forge
openssl               	3.0.8            	h0b41bf4_0	conda-forge
opt-einsum            	3.3.0                	pypi_0	pypi
packaging             	23.0           	pyhd8ed1ab_0	conda-forge
pandas                	1.5.3                	pypi_0	pypi
parso                 	0.8.3          	pyhd8ed1ab_0	conda-forge
partd                 	1.3.0          	pyhd8ed1ab_0	conda-forge
pcre2                 	10.40            	hc3806b6_0	conda-forge
pexpect               	4.8.0          	pyh1a96a4e_2	conda-forge
pickleshare           	0.7.5               	py_1003	conda-forge
pillow                	9.4.0        	py38hde6dc18_1	conda-forge
pint                  	0.20.1         	pyhd8ed1ab_0	conda-forge
pip                   	23.0.1         	pyhd8ed1ab_0	conda-forge
pixman                	0.40.0           	h36c2ea0_0	conda-forge
pkgutil-resolve-name  	1.3.10         	pyhd8ed1ab_0	conda-forge
platformdirs          	3.0.0          	pyhd8ed1ab_0	conda-forge
ply                   	3.11                   	py_1	conda-forge
pooch                 	1.7.0          	pyhd8ed1ab_0	conda-forge
prompt-toolkit        	3.0.38         	pyha770c72_0	conda-forge
prompt_toolkit        	3.0.38           	hd8ed1ab_0	conda-forge
protobuf              	3.19.6               	pypi_0	pypi
psutil                	5.9.4        	py38h0a891b7_0	conda-forge
psygnal               	0.7.0                	pypi_0	pypi
pthread-stubs         	0.4           	h36c2ea0_1001	conda-forge
ptyprocess            	0.7.0          	pyhd3deb0d_0	conda-forge
pulseaudio            	16.1             	ha8d29e2_1	conda-forge
pure_eval             	0.2.2          	pyhd8ed1ab_0	conda-forge
pyasn1                	0.4.8                	pypi_0	pypi
pyasn1-modules        	0.2.8                	pypi_0	pypi
pycparser             	2.21           	pyhd8ed1ab_0	conda-forge
pydantic              	1.10.5       	py38h1de0b5d_0	conda-forge
pygments              	2.14.0         	pyhd8ed1ab_0	conda-forge
pyopengl              	3.1.6          	pyhd8ed1ab_1	conda-forge
pyopenssl             	23.0.0         	pyhd8ed1ab_0	conda-forge
pyparsing             	3.0.9                	pypi_0	pypi
pyproject_hooks       	1.0.0          	pyhd8ed1ab_0	conda-forge
pyqt5                 	5.12.3               	pypi_0	pypi
pyqt5-qt5             	5.15.2               	pypi_0	pypi
pyqt5-sip             	12.11.1              	pypi_0	pypi
pyrsistent            	0.19.3       	py38h1de0b5d_0	conda-forge
pysocks               	1.7.1          	pyha2e5f31_6	conda-forge
python                	3.8.16      	he550d4f_1_cpython	conda-forge
python-build          	0.10.0         	pyhd8ed1ab_0	conda-forge
python-dateutil       	2.8.2          	pyhd8ed1ab_0	conda-forge
python_abi            	3.8                  	3_cp38	conda-forge
pytomlpp              	1.0.11       	py38h43d8883_1	conda-forge
pytz                  	2022.7.1       	pyhd8ed1ab_0	conda-forge
pywavelets            	1.4.1        	py38h7e4f40d_0	conda-forge
pyyaml                	6.0          	py38h0a891b7_5	conda-forge
pyzmq                 	25.0.0       	py38he24dcef_0	conda-forge
qt-main               	5.15.8           	h5d23da1_6	conda-forge
qtconsole-base        	5.4.0          	pyha770c72_0	conda-forge
qtpy                  	2.3.0          	pyhd8ed1ab_0	conda-forge
readline              	8.1.2            	h0f457ee_0	conda-forge
requests              	2.28.2         	pyhd8ed1ab_0	conda-forge
requests-oauthlib     	1.3.1                	pypi_0	pypi
rich                  	13.3.1         	pyhd8ed1ab_1	conda-forge
rsa                   	4.9                  	pypi_0	pypi
scikit-image          	0.20.0               	pypi_0	pypi
scipy                 	1.9.1                	pypi_0	pypi
setuptools            	67.4.0         	pyhd8ed1ab_0	conda-forge
shellingham           	1.5.1          	pyhd8ed1ab_0	conda-forge
sip                   	6.7.7        	py38h8dc9893_0	conda-forge
six                   	1.16.0         	pyh6c4a22f_0	conda-forge
snappy                	1.1.9            	hbd366e4_2	conda-forge
snowballstemmer       	2.2.0          	pyhd8ed1ab_0	conda-forge
sortedcontainers      	2.4.0          	pyhd8ed1ab_0	conda-forge
sphinx                	4.5.0                	pypi_0	pypi
sphinxcontrib-applehelp   1.0.4          	pyhd8ed1ab_0	conda-forge
sphinxcontrib-devhelp 	1.0.2                  	py_0	conda-forge
sphinxcontrib-htmlhelp	2.0.1          	pyhd8ed1ab_0	conda-forge
sphinxcontrib-jsmath  	1.0.1                  	py_0	conda-forge
sphinxcontrib-qthelp  	1.0.3                  	py_0	conda-forge
sphinxcontrib-serializinghtml 1.1.5          	pyhd8ed1ab_2	conda-forge
stack_data            	0.6.2          	pyhd8ed1ab_0	conda-forge
stardist              	0.8.3                	pypi_0	pypi
stardist-napari       	2022.12.6            	pypi_0	pypi
superqt               	0.4.1          	pyhd8ed1ab_0	conda-forge
tblib                 	1.7.0          	pyhd8ed1ab_0	conda-forge
tensorboard           	2.11.2               	pypi_0	pypi
tensorboard-data-server   0.6.1                	pypi_0	pypi
tensorboard-plugin-wit	1.8.1                	pypi_0	pypi
tensorflow            	2.11.0               	pypi_0	pypi
tensorflow-estimator  	2.11.0               	pypi_0	pypi
tensorflow-io-gcs-filesystem 0.31.0               	pypi_0	pypi
termcolor             	2.2.0                	pypi_0	pypi
tifffile              	2023.2.28            	pypi_0	pypi
tk                    	8.6.12           	h27826a3_0	conda-forge
toml                  	0.10.2         	pyhd8ed1ab_0	conda-forge
tomli                 	2.0.1          	pyhd8ed1ab_0	conda-forge
toolz                 	0.12.0         	pyhd8ed1ab_0	conda-forge
tornado               	6.2          	py38h0a891b7_1	conda-forge
tqdm                  	4.64.1         	pyhd8ed1ab_0	conda-forge
traitlets             	5.9.0          	pyhd8ed1ab_0	conda-forge
typer                 	0.7.0                	pypi_0	pypi
typing-extensions     	4.5.0                	pypi_0	pypi
typing_extensions     	4.4.0          	pyha770c72_0	conda-forge
urllib3               	1.26.14        	pyhd8ed1ab_0	conda-forge
vispy                 	0.11.0               	pypi_0	pypi
wcwidth               	0.2.6          	pyhd8ed1ab_0	conda-forge
werkzeug              	2.2.3                	pypi_0	pypi
wheel                 	0.38.4         	pyhd8ed1ab_0	conda-forge
wrapt                 	1.15.0       	py38h1de0b5d_0	conda-forge
xcb-util              	0.4.0            	h166bdaf_0	conda-forge
xcb-util-image        	0.4.0            	h166bdaf_0	conda-forge
xcb-util-keysyms      	0.4.0            	h166bdaf_0	conda-forge
xcb-util-renderutil   	0.3.9            	h166bdaf_0	conda-forge
xcb-util-wm           	0.4.1            	h166bdaf_0	conda-forge
xorg-kbproto          	1.0.7         	h7f98852_1002	conda-forge
xorg-libice           	1.0.10           	h7f98852_0	conda-forge
xorg-libsm            	1.2.3         	hd9c2040_1000	conda-forge
xorg-libx11           	1.8.4            	h0b41bf4_0	conda-forge
xorg-libxau           	1.0.9            	h7f98852_0	conda-forge
xorg-libxdmcp         	1.1.3            	h7f98852_0	conda-forge
xorg-libxext          	1.3.4            	h0b41bf4_2	conda-forge
xorg-libxrender       	0.9.10        	h7f98852_1003	conda-forge
xorg-renderproto      	0.11.1        	h7f98852_1002	conda-forge
xorg-xextproto        	7.3.0         	h0b41bf4_1003	conda-forge
xorg-xproto           	7.0.31        	h7f98852_1007	conda-forge
xz                    	5.2.6            	h166bdaf_0	conda-forge
yaml                  	0.2.5            	h7f98852_2	conda-forge
yarl                  	1.8.2        	py38h0a891b7_0	conda-forge
zarr                  	2.13.6         	pyhd8ed1ab_0	conda-forge
zeromq                	4.3.4            	h9c3ff4c_1	conda-forge
zfp                   	1.0.0            	h27087fc_3	conda-forge
zict                  	2.2.0          	pyhd8ed1ab_0	conda-forge
zipp                  	3.15.0         	pyhd8ed1ab_0	conda-forge
zlib                  	1.2.13           	h166bdaf_4	conda-forge
zlib-ng               	2.0.6            	h166bdaf_0	conda-forge
zstd                  	1.5.2            	h3eb15da_6	conda-forge
```

**Comment by tlambert03** [comment]
thanks a lot.  I'm at a hackathon at the moment, but will try to get a fix released in the next day or so

**Comment by tlambert03** [comment]
ok, I was able to figure out the issue.
It's not really a psygnal problem per se, but it's another example of the difficulty of using pip and conda together to control the same packages in the same environment.

Specifically, this happens when you **pip** install psygnal, and then *downgrade* psygnal using **conda** (which, probably unbeknownst to you, happened when you ran `conda install napari-ome-zarr`)

```bash
$ conda create -n test python=3.8
$ conda activate test

$ pip install psygnal  # gets latest psygnal (0.8.1) via pip
$  python -c "import psygnal"  # works

$ conda install -c conda-forge psygnal==0.7.2  # downgrading psygnal with conda
$ python -c "import psygnal"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/talley/mambaforge/envs/test/lib/python3.8/site-packages/psygnal/__init__.py", line 58, in <module>
    from ._evented_decorator import evented, get_evented_namespace, is_evented
ImportError: cannot import name 'get_evented_namespace' from 'psygnal._evented_decorator' (/Users/talley/mambaforge/envs/test/lib/python3.8/site-packages/psygnal/_evented_decorator.cpython-38-darwin.so)

$ pip install psygnal==0.7.2  # overriding the conda version with the pip version
$  python -c "import psygnal"  # works
```

the reason for this is that psygnal 0.8.1 has *compiled* files (for performance reasons), and when conda downgraded it to 0.7.2, it didn't properly clean up the compiled files.  So upon import, you get a strange mismatch of files that just don't work.

The reason conda downgraded it was that, until just now, 0.7.2 was the latest version on conda... (0.8.0 was just recently released and I just pushed it to conda, so in an hour or so this probably won't happen again)

I'm afraid there  isn't much I can do here to prevent the fact that, in rare occasions like this, conda and pip don't always play perfectly well together when there are compiled files involved.  In your environment setup above, I see this:

1. `create a napari conda env with python 3.8`
2. `pip install napari[all]` - All is well, I can import napari and psygnal
3. `conda install -n napari -c conda-forge napari-ome-zarr` - here's where psygnal gets downgraded and you have breakage
4. `pip install psygnal==0.7.2` ... it wasn't the specific version that fixed you here, it was the fact that you installed with pip.


In the future, I'd also suggest trying to stick with one package manager if possible (though it very often *also* works fine to mix).  For example, rather than the multi-step install you used above, how about:

```bash
mamba create -n napari python=3.8 napari pyqt==5.12.3 napari-ome-zarr cudatoolkit=11.2 stardist-napari
```
